### PR TITLE
[baseline][libzim][xapian] Disable gtest, fix mingw and windows builds

### PR DESCRIPTION
**Pull request by dg0yt**
# `xapian`

Amends #31338:
Use pkgconfig for zlib. Fixes mingw and debug builds
Really disable the `_DEBUG` macro check which doesn't work for vcpkg's single includes installation.

# `libzim`

Amends #30758:
Fixes baseline regression: Don't pick gtest when it is installed before libzim. (Building the tests doesn't work and isn't desired in vcpkg.)
Fix mingw and cross builds.
Fix dllexport/dllimport.
Enable feature xapian for windows triplets.

# Integration

`libzim[xapian]` tested with windows triplets in https://dev.azure.com/vcpkg/public/_build/results?buildId=89372.

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by FrankXie05** [comment]
Could you please describe this `baseline regression`?

**Comment by dg0yt** [comment]
> Could you please describe this `baseline regression`?

See the first commit and its CI run: if gtest is installed before libzim, it is picked up and leads to a build error.

File: ports/xapian/portfile.cmake [hunk 1]
```diff
@@ -1,43 +1,34 @@
...
     ARCHIVE "${ARCHIVE}"
+    PATCHES
+        configure.diff
+        msvc-no-debug.diff
 )
 
-if(WIN32)
-    vcpkg_replace_string("${SOURCE_PATH}/configure.ac" "z zlib zdll" "z zlib zdll zlibd")
-
-    if(MSVC)
```
Comment (dg0yt) [inline ports/xapian/portfile.cmake:15]: NB: WIN32 and MSVC are not defined in script mode. This section was never entered.

File: ports/libzim/dllexport.diff [hunk 2]
```diff
@@ -0,0 +1,52 @@
+diff --git a/include/zim/zim.h b/include/zim/zim.h
+index 5d1af80..6171f56 100644
+--- a/include/zim/zim.h
++++ b/include/zim/zim.h
+@@ -33,14 +33,16 @@
+ #define DEPRECATED
+ #endif
+ 
+-#if (defined _WIN32 || defined __CYGWIN__) && defined LIBZIM_EXPORT_DLL
```
Comment (dg0yt) [inline ports/libzim/dllexport.diff:9]: AFAICT the syntax is `defined(...)`, not `defined ...`.

File: ports/libzim/portfile.cmake [hunk 3]
```diff
@@ -14,20 +16,14 @@ if(NOT "xapian" IN_LIST FEATURES)
     list(APPEND EXTRA_OPTIONS "-Dwith_xapian=false")
 endif()
 
-if(VCPKG_LIBRARY_LINKAGE STREQUAL dynamic)
-    list(APPEND EXTRA_OPTIONS "-Dstatic-linkage=false")
-else()
-    list(APPEND EXTRA_OPTIONS "-Dstatic-linkage=true")
-endif()
```
Comment (dg0yt) [inline ports/libzim/portfile.cmake:21]: It follows the `default_library` setting which is already given by vcpkg.

File: scripts/ci.baseline.txt [hunk 4]
```diff
@@ -1713,10 +1713,7 @@ x265:arm64-windows=fail
 x265:arm-uwp=fail
 x265:x64-android=fail
 x265:x64-uwp=fail
-xalan-c:x64-windows-static=fail
-xalan-c:arm64-windows=fail
```
Comment (dg0yt) [inline scripts/ci.baseline.txt:1717]: FTR xalan-c was delisted so these records are obsolete.

**Review by FrankXie05 — APPROVED** [review]

File: ports/libzim/dllexport.diff (hunk 2 above)
Comment (xiaoyifang) [reply ports/libzim/dllexport.diff:9]: I think they have the same effect with differnt syntax

Comment (dg0yt) [reply ports/libzim/dllexport.diff:9]: Okay, `defined MACRO` is possible, too.

**Pull request merged**